src/core/or/connection_or.c

```diff
@@ -1516,13 +1524,15 @@ connection_or_connect, (const tor_addr_t *_addr, uint16_t port,
   conn->is_outgoing = 1;
 
   /* If we are using a proxy server, find it and use it. */
-  r = get_proxy_addrport(&proxy_addr, &proxy_port, &proxy_type, TO_CONN(conn));
+  r = get_proxy_addrport(&proxy_addr, &proxy_port, &proxy_type, &is_pt,
+                         TO_CONN(conn));
   if (r == 0) {
     conn->proxy_type = proxy_type;
     if (proxy_type != PROXY_NONE) {
       tor_addr_copy(&addr, &proxy_addr);
       port = proxy_port;
       conn->base_.proxy_state = PROXY_INFANT;
+      conn->is_pt = is_pt;
```

Comment:
We should either use the "!!" trick here to force "is_pt" to 0 or 1, or we should document in get_proxy_addrport that is_pt will be set to 0 or 1.

Reply:
I'd rather go the documentation route. We could also have `is_pt` be a `bool`, but that conflicts with surrounding practice.

Reply:
ok